src/Engine/Scene/SkinningComponent.hpp

```diff
@@ -0,0 +1,246 @@
+ * the following data from the skinned mesh's name:
+ *    - the Ra::Core::Animation::SkinningRefData containing all the
+ *      reference data needed for the skinning.
+ *    - the Ra::Core::Animation::SkinningFrameData containing all the
+ *      data needed for the skinning at the current frame.
+ *
+ * \warning The Ra::Core::Animation::SkinningFrameData better be accessed after
+ *          the skinning task.
+ */
+class RA_ENGINE_API SkinningComponent : public Component
+{
+  public:
+    /// The Geometric Skinning Method.
+    enum SkinningType {
+        LBS = 0,  ///< Linear Blend Skinning
+        DQS,      ///< Dual Quaternion Skinning
+        COR,      ///< Center of Rotation skinning
+        STBS_LBS, ///< Stretchable Twistable Bone Skinning with LBS
+        STBS_DQS  ///< Stretchable Twistable Bone Skinning with DQS
+    };
+
+    /// How to skin the normal, tangent and binormal vectors.
+    enum NormalSkinning {
+        APPROX = 0, ///< Use the standard approximation method.
+        GEOMETRIC   ///< Recompute from scratch from the skinned positions.
+    };
+
+    /// The skinning weight type.
+    enum WeightType {
+        STANDARD = 0, ///< Standard geometric skinning weights
+        STBS          ///< Stretchable Twistable Bone Skinning weights
+    };
+
+    SkinningComponent( const std::string& name, SkinningType type, Entity* entity ) :
+        Component( name, entity ),
+        m_skinningType( type ),
+        m_normalSkinning( APPROX ),
+        m_isReady( false ),
+        m_forceUpdate( false ),
+        m_weightBone( 0 ),
+        m_weightType( STANDARD ),
+        m_showingWeights( false ) {}
+
+    ~SkinningComponent() override {}
+
+    /// \name Component Communication (CC)
+    /// \{
+
+    virtual void initialize() override;
+    /// \}
+
+    /// \name Build from fileData
+    /// \{
+
+    /// Loads the skinning data from the given Handledata.
+    /// \note Call initialize() afterwards to finalize data registration.
+    void handleSkinDataLoading( const Core::Asset::HandleData* data, const std::string& meshName );
+    /// \}
+
+    /// \name Skinning Process
+    /// \{
+
+    /// Apply the Skinning Method and update the SkinningFrameData.
+    void skin();
+
+    /// Update internal data and update the skinned mesh.
+    void endSkinning();
+
+    /// Sets the Skinning method to use.
+    void setSkinningType( SkinningType type );
+
+    /// Returns the current skinning method.
+    inline SkinningType getSkinningType() const { return m_skinningType; }
+
+    /// Sets the method to use to skin the normal, tangent and binormal vectors.
+    void setNormalSkinning( NormalSkinning normalSkinning );
+
+    /// Returns the current method used to skin the normal, tangent and binormal vectors.
+    inline NormalSkinning getNormalSkinning() const { return m_normalSkinning; }
+
+    /// Toggles use of smart stretch.
+    void setSmartStretch( bool on );
```

Comment:
Shall we switch the name from `smart stretch` to `pseudo IK` ?

Reply:
todo added on the code. Will be done in a next PR